op-e2e/actions/helpers/l1_miner.go

```diff
@@ -177,8 +177,11 @@ func (s *L1Miner) IncludeTx(t Testing, tx *types.Transaction) {
 		return
 	}
 	s.l1BuildingState.SetTxContext(tx.Hash(), len(s.L1Transactions))
-	receipt, err := core.ApplyTransaction(s.l1Cfg.Config, s.l1Chain, &s.l1BuildingHeader.Coinbase,
-		s.L1GasPool, s.l1BuildingState, s.l1BuildingHeader, tx.WithoutBlobTxSidecar(), &s.l1BuildingHeader.GasUsed, *s.l1Chain.GetVMConfig())
+	blockCtx := core.NewEVMBlockContext(s.l1BuildingHeader, s.l1Chain, nil, s.l1Cfg.Config, s.l1BuildingState)
+	// TODO: we could reuse the evm for the same block and use SetTxContext here
+	evm := vm.NewEVM(blockCtx, s.l1BuildingState, s.l1Cfg.Config, *s.l1Chain.GetVMConfig())
+	receipt, err := core.ApplyTransaction(
+		evm, s.L1GasPool, s.l1BuildingState, s.l1BuildingHeader, tx.WithoutBlobTxSidecar(), &s.l1BuildingHeader.GasUsed)
```

Comment:
Wondering if the txContext has to be set here too. Old code inside `core.ApplyTransaction` was
```
	msg, err := TransactionToMessage(tx, types.MakeSigner(config, header.Number, header.Time), header.BaseFee)
	if err != nil {
		return nil, err
	}
	txContext := NewEVMTxContext(msg)
```

Reply:
See other comment, the `ApplyTransaction` still sets tx context.